### PR TITLE
Cloud serve

### PR DESCRIPTION
**Pull request by neosbob**

### CODE REVIEW

**Comment by neosbob** [comment]
Feel free to point out any issues with my code. Going to fix the conflict with Makefile after I read through the differences.

File: cloud_file_handler.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#include <sstream>
+#include <cstdlib>
+#include <utility>
+#include <map>
+#include "request_handler.h"
+#include "not_found_handler.h"
+
+typedef std::vector<std::pair<std::string, std::string>> Headers;
+
+// Represents the parent of all request handlers. Implementations should expect to
```
Comment (echen94) [inline cloud_file_handler.h]: Can change the comment here to describe what cloud_file_handler does.

File: cloud_file_handler.cc [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    else if (request_path == "/")
+    {
+        return RequestHandler::Status::FILE_NOT_FOUND;     
+    }
+    else 
+    {
+        return RequestHandler::Status::FILE_NOT_FOUND;
+    }
+}
+
```
Comment (echen94) [inline cloud_file_handler.cc:122]: Can add comment on how this function works.

File: cloud_file_handler.cc [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+RequestHandler::Status CloudHandler::Init(const std::string& uri_prefix, const NginxConfig& config)
+{   
+    bool remote_host_found = false;
+
+    if (config.statements_[0]->tokens_[0] == "root" && config.statements_[0]->tokens_.size() == 2)
+    {
+        root_dir_ = config.statements_[0]->tokens_[1];
+        uri_prefix_ = uri_prefix; //in this case "/cloud"
+	//remote_host_found = parse_remote_url(root_dir_);
```
Comment (echen94) [inline cloud_file_handler.cc:31]: can delete commented code.

File: cloud_file_handler.h [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  // HTTP code 500.
+  virtual Status HandleRequest(const Request& request,
+                               Response* response);
+  std::string get_cloud_object(std::string remote_dir, std::string remote_file_path);
+  bool parse_remote_response(std::string remote_response);
+  bool read_header(std::string headers);
+  //bool parse_remote_url(std::string remote_host_url);
+
+  private:
+        std::string root_dir_;
```
Comment (echen94) [inline cloud_file_handler.h]: minor issue in indentation

**Comment by echen94** [comment]
Good integration test and good cleanup of old files!

**Pull request merged**